### PR TITLE
fix(server): kneigbor-api has unmatched edge type with server

### PR DESCRIPTION
**Pull request by LiJie20190102**
## Purpose of the PR

- close https://github.com/apache/incubator-hugegraph-toolchain/pull/644

## Verifying these changes

- [ ] Trivial rework / code cleanup without any test coverage. (No Need)
- [ ] Already covered by existing tests, such as *(please modify tests here)*.
- [ ] Need tests and can be verified as follows:
```
graphs/graph_demo1/traversers/kneighbor
    {
	"countOnly": false,
	"limit": 10000000,
	"maxDepth": 1,
	"pathWithEdge": false,
	"steps": {
		"direction": "BOTH",
		"maxDegree": 1000,
		"skipDegree": 0,
		"vSteps": []
	},
	"withEdge": false,
	"withPath": true,
	"withVertex": false
}


**result:**

{
	"kneighbor": [
		"2:car1!2",
		"2:car2!4"
	],
	"size": 2,
	"paths": [
		{
			"objects": [
				"1:zhangsan",
				"2:car1!2"
			]
		},
		{
			"objects": [
				"1:zhangsan",
				"2:car2!4"
			]
		}
	],
	"vertices": [],
	"edges": []
}
```

```
graphs/graph_demo1/traversers/kneighbor
    {
	"countOnly": false,
	"limit": 10000000,
	"maxDepth": 1,
	"pathWithEdge": false,
	"steps": {
		"direction": "BOTH",
		"maxDegree": 1000,
		"skipDegree": 0,
		"vSteps": []
	},
	"withEdge": true,
	"withPath": true,
	"withVertex": true
}


**result:**

{
	"kneighbor": [
		"2:car1!2",
		"2:car2!4"
	],
	"size": 2,
	"paths": [
		{
			"objects": [
				"1:zhangsan",
				"2:car1!2"
			]
		},
		{
			"objects": [
				"1:zhangsan",
				"2:car2!4"
			]
		}
	],
	"vertices": [
		{
			"id": "2:car2!4",
			"label": "car",
			"type": "vertex",
			"properties": {
				"age": 4,
				"qsdi_task_id": "car2_task_id_test",
				"qsdi_record_id": "test_record_id",
				"name": "car2",
				"qsdi_data_source": "1"
			}
		},
		{
			"id": "1:zhangsan",
			"label": "people",
			"type": "vertex",
			"properties": {
				"age": 55,
				"qsdi_record_id": "926133154860048384",
				"name": "zhangsan",
				"qsdi_data_source": "1"
			}
		},
		{
			"id": "2:car1!2",
			"label": "car",
			"type": "vertex",
			"properties": {
				"age": 2,
				"qsdi_task_id": "car1_task_id_test",
				"qsdi_record_id": "car1_log_id_test",
				"name": "car1",
				"qsdi_data_source": "1"
			}
		}
	],
	"edges": [
		{
			"id": "S1:zhangsan>3>3>>S2:car2!4",
			"label": "driver_single",
			"type": "edge",
			"outV": "1:zhangsan",
			"outVLabel": "people",
			"inV": "2:car2!4",
			"inVLabel": "car",
			"properties": {
				"qsdi_task_id": "zhangsan_car2_rask_id_test",
				"qsdi_record_id": "926133802515111937",
				"name": "ride12",
				"driver_count": 23,
				"qsdi_data_source": "1"
			}
		},
		{
			"id": "S1:zhangsan>3>3>>S2:car1!2",
			"label": "driver_single",
			"type": "edge",
			"outV": "1:zhangsan",
			"outVLabel": "people",
			"inV": "2:car1!2",
			"inVLabel": "car",
			"properties": {
				"qsdi_record_id": "926133804008284160",
				"name": "ride88",
				"driver_count": 88,
				"qsdi_data_source": "1"
			}
		}
	]
}
```

## Does this PR potentially affect the following parts?

- [ ]  Dependencies ([add/update license](https://hugegraph.apache.org/docs/contribution-guidelines/contribute/#321-check-licenses) info & [regenerate_known_dependencies.sh](../install-dist/scripts/dependency/regenerate_known_dependencies.sh)) 
- [ ]  Modify configurations
- [ ]  The public API
- [ ]  Other affects (typed here)
- [ ]  Nope

## Documentation Status

- [ ]  `Doc - TODO` 
- [ ]  `Doc - Done` 
- [ ]  `Doc - No Need`

### CODE REVIEW

**Comment by LiJie20190102** [comment]
@imbajin  Please help review it

**Review by imbajin — APPROVED** [review]
Good to see the tests meets expectations

File: hugegraph-server/hugegraph-api/src/main/java/org/apache/hugegraph/api/traversers/KneighborAPI.java [hunk 1]
```diff
@@ -179,18 +180,12 @@ public String post(@Context GraphManager manager,
...
             iterVertex = g.vertices(vertexIds.toArray());
             measure.addIterCount(vertexIds.size(), 0L);
-        } else {
-            iterVertex = vertexIds.iterator();
         }
 
-        Iterator<?> iterEdge = Collections.emptyIterator();
-        if (request.withPath) {
+        Iterator<Edge> iterEdge = Collections.emptyIterator();
+        if (request.withPath && request.withEdge) {
```
Comment (javeme) [inline hugegraph-server/hugegraph-api/src/main/java/org/apache/hugegraph/api/traversers/KneighborAPI.java:186]: looks like we cut the returning ids branch?
due to it is not easy for the client to determine whether it is an edge id or an edge object?

**Review by javeme — APPROVED** [review]

**Comment by javeme** [comment]
@LiJie20190102  Thank you for your contribution. please also change the api version to 0.72: https://github.com/apache/incubator-hugegraph/blob/master/hugegraph-server/hugegraph-api/src/main/java/org/apache/hugegraph/version/ApiVersion.java#L123

**Pull request merged**